lib/ansiblelint/__main__.py

```diff
@@ -56,6 +57,20 @@ def initialize_logger(level: int = 0) -> None:
     _logger.debug("Logging initialized to level %s", logging_level)
 
 
+def initalize_formatter_factory(options_list: Namespace):
```

Comment:
This name is misleading. Classes have `__init__()` method and this is what's called an initializer because it performs the initialization of the class state post-creation (when you call `ClassName(...)` it first gets instantiated via `__new__()` constructor and then initialized via `__init__()`. Initializing is done after this function is called.

It would be appropriate to use something along the lines of "select", or "choose", or "pick" maybe...

Also, regarding `options_list`. It's probably not exactly a list so maybe invent some better var name too. Like `cli_args`.

```suggestion
def choose_formatter_factory(options_list: Namespace) -> Type[formatters.BaseFormatter]:
```